scalyr_agent/builtin_monitors/linux_process_metrics.py

```diff
@@ -834,19 +832,19 @@ def _initialize(self):
         self.__include_child_processes = self._config.get(
             'include_child_processes', default=False, convert_to=bool
         )
-        if not getattr(self, 'last_discovered', False):
+
+        self.__trackers = {}  # key -> process id, value -> ProcessTracker object
+        if not getattr(self, 'last_discovered', None):
             # poll interval (in seconds) for matching processes when we are monitoring multiple processes
             # alive for all epochs of the monitor.
-            self.__last_discovered = {}
+            self.__last_discovered = None
         self.__process_discovery_interval = self._config.get('process_discovery_interval', default=120, convert_to=int)
-        self.__datetime_format = "%Y-%m-%d %H:%M:%S"
-
-        # this is the key to the dictionary self.__last_discovered which denotes the key to the monitor process
-        self.__last_polled_env_key = '%s_linux_process_monitor_last_polled' % self.__id
 
         __target_pids = self._config.get('pid', default=None, convert_to=str)
         self.__target_pids = []
 
+        self.__running_pids = []  # list of all process ids that are currently running
```

Comment:
How is this different than `self.__pids`?  I think we can remove `self.__running_pids`.

Reply:
`__running_pids` is a list of *all* the running processes at a particular time. `__pids` is the list of running processes we are interested in.

Reply:
but you're right. it can be refactored and we don't really need `__running_pids`